Lesson2/src/main/java/pl/slvd/university/Deanery.java

```diff
@@ -0,0 +1,43 @@
+package pl.slvd.university;
+
+import java.util.*;
+import java.util.stream.IntStream;
+
+class Deanery implements Comparator<Applicant> {
+
+    @Override
+    public int compare(Applicant o1, Applicant o2) {
+        return o2.getSum() - (o1.getSum());
+    }
+
+    public static void sortByGrades(List<Applicant> sortedList, String firstLastName, int getNumOfBudgetPlaces, int getNumOfPaidPlaces) {
+        System.out.println("Ready to see the lists of enrolled in the University?");
+        Scanner scanner = new Scanner(System.in);
+        String answer = scanner.next().toUpperCase();
+        switch (answer) {
+            case "NO" -> sortByGrades(sortedList, firstLastName, getNumOfBudgetPlaces, getNumOfPaidPlaces);
+            case "YES" -> {
+                Set<Applicant> sortedApplicants = new TreeSet<>(new Deanery());
+                sortedApplicants.addAll(sortedList);
+                List<Applicant> sortedApplicantsCopy = sortedApplicants.stream().toList();
+
+                IntStream.range(0, sortedApplicantsCopy.size()).forEach(i ->
+                        System.out.println((i + 1) + ". " + sortedApplicantsCopy.get(i)));
+                int sum = getNumOfBudgetPlaces + getNumOfPaidPlaces;
+
+                for (int i = 0; i < sortedApplicantsCopy.size(); i++) {
+                    if (sortedApplicantsCopy.get(i).getFirstLastName().equals(firstLastName) && (i >= sum)) {
+                        System.out.println("Unfortunately, you are not enrolled in the University.");
+                        break;
+                    } else if (sortedApplicantsCopy.get(i).getFirstLastName().equals(firstLastName) && (i <= getNumOfBudgetPlaces - 1)) {
+                        System.out.println("Great! You are enrolled in the University for a budget place!");
+                        break;
+                    } else if (sortedApplicantsCopy.get(i).getFirstLastName().equals(firstLastName) && (i > getNumOfBudgetPlaces - 1) && (i <= (sum - getNumOfBudgetPlaces))) {
+                        System.out.println("Great! You are enrolled in paid education at the University! Your studies will cost $2,000 per year.");
+                        break;
+                    }
+                }
```

Comment:
Since the first condition is similar for every if, we can do smth like this:
```
                for (int i = 0; i < sortedApplicantsCopy.size(); i++) {
                    if (sortedApplicantsCopy.get(i).getFirstLastName().equals(firstLastName)) {
                        if (i >= sum) {
                            System.out.println("Unfortunately, you are not enrolled in the University.");
                            break;
                        } else if (i <= getNumOfBudgetPlaces - 1) {
                            System.out.println("Great! You are enrolled in the University for a budget place!");
                            break;
                        } else if (i > getNumOfBudgetPlaces - 1) && (i <= (sum - getNumOfBudgetPlaces)) {
                            System.out.println("Great! You are enrolled in paid education at the University! Your studies will cost $2,000 per year.");
                            break;
                        }
                    }
                }
```

Reply:
In this case we do the first check only once, and in case it doesn't work, we proceed to the next applicant. In your case for the same condition this check will be done 3 times in case applicant is not in a list.

Reply:
Ok!